src/pages/subjects/Subjects.tsx

```diff
@@ -125,6 +122,13 @@ const Subjects = () => {
     />
   )
 
+  const options = useMemo(
+    () => subjectsNamesItems.map((option: SubjectNameInterface) => option.name),
+    [subjectsNamesItems]
+  )
+
+  const getSubjectNames = () => !subjectsNamesItems.length && void fetchData()
```

Comment:
if I chose another category i wont get subjects to it if I already have fetched subjects

Reply:
> if I chose another category i wont get subjects to it if I already have fetched subjects

fixed it